utils/generic.py

```diff
@@ -837,10 +881,20 @@ def assign_population_pcs(
     :param float min_prob: Minimum probability of belonging to a given population for the population to be set (otherwise set to `None`)
     :param str output_col: Output column storing the assigned population
     :param str missing_label: Label for samples for which the assignment probability is smaller than `min_prob`
-    :return: Dataframe containing sample IDs and imputed population labels, trained random forest model
-    :rtype: DataFrame, RandomForestClassifier
+    :return: Hail Table or Pandas Dataframe (depending on input) containing sample IDs and imputed population labels, trained random forest model
+    :rtype: (Table or DataFrame, RandomForestClassifier)
     """
 
+    hail_input = isinstance(pop_pca_scores, hl.Table)
+    if hail_input:
+        pop_pc_pd = pop_pca_scores.select(
+            known_col,
+            pca_scores=pc_cols
+        ).to_pandas()
+        pop_pc_pd = expand_pd_array_col(pop_pc_pd, 'pca_scores', out_cols_prefix='PC')
```

Comment:
So, you  can also use the expanded version form a hail table; this should work:
`assign_population_pcs(ht, pc_cols=[ht.pc1, ht.pc2, ...])`

Now , you're right that the reverse isn't true for the Pandas version (it has to be in different columns). 

The main reason for having it this way is because it doesn't break any legacy code  while adding the desired functionality. If you think it's worthwhile, I _could_ make if work for Pandas where PCs are in a single column.

Reply:
I think you are right that it makes sense to do it this way to work with the older code too. I think I was just getting stuck thinking about it for a bit.